src/reducers/subreddits.test.js

```diff
@@ -0,0 +1,174 @@
+    threads: [{
+      data: {
+        id: 'baz',
+      },
+    }],
+    filter: 'foo',
+    subreddit: 'bar',
+  });
+
+  const bar = t.context.state.getIn(['threads', 'bar']);
+  t.true(bar.has('baz'), 'should have the given thread');
+  t.deepEqual(bar.getIn(['baz', 'filters']).toArray(), ['foo'], 'should have the correct filter');
+});
+
+test('SAVE_THREAD_COMMENTS should set the comments correctly', t => {
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.SAVE_THREAD_COMMENTS, {
+    comments: [{
+      data: {
+        id: 'baz',
+      },
+    }],
+    thread: 'foo',
+    subreddit: 'bar',
+  });
+
+  t.true(t.context.state.getIn(['comments', 'bar', 'foo', 'baz']).has('data'));
+});
+
+test('SAVE_THREAD_COMMENTS_TO_CACHE should set the comment cache correctly', t => {
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.SAVE_THREAD_COMMENTS_TO_CACHE, {
+    comments: [{
+      data: {
+        id: 'baz',
+      },
+    }],
+    thread: 'foo',
+    subreddit: 'bar',
+  });
+
+  t.true(t.context.state.getIn(['subredditCommentCache', 'bar', 'foo', 'baz']).has('data'));
+});
+
+test('TOGGLE_THREAD_COMMENT_EXPAND_CHILDREN should properly toggle', t => {
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.TOGGLE_THREAD_COMMENT_EXPAND_CHILDREN, {
+    comment: 'hello',
+    thread: 'foo',
+    subreddit: 'bar',
+  });
+  t.false(
+    t.context.state.getIn(['expandChildren', 'bar', 'foo', 'hello']),
+    'should be toggled off since it was true by default'
+  );
+
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.TOGGLE_THREAD_COMMENT_EXPAND_CHILDREN, {
+    comment: 'hello',
+    thread: 'foo',
+    subreddit: 'bar',
+  });
+  t.true(
+    t.context.state.getIn(['expandChildren', 'bar', 'foo', 'hello']),
+    'should be toggled back on'
+  );
+});
+
+test('DELETE_THREAD_COMMENT should remove the given comment', t => {
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.SAVE_THREAD_COMMENTS, {
+    comments: [{
+      data: {
+        id: 'baz',
+      },
+    }],
+    thread: 'bar',
+    subreddit: 'foo',
+  });
+  t.true(t.context.state.getIn(['comments', 'foo', 'bar', 'baz']).has('data'));
+
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.DELETE_THREAD_COMMENT, {
+    subreddit: 'foo',
+    thread: 'bar',
+    comment: 'baz',
+  });
+  t.false(t.context.state.getIn(['comments', 'foo', 'bar']).has('baz'));
+});
+
+test('TOGGLE_THREAD_CARD_EXPAND_THUMBNAIL should properly toggle', t => {
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.TOGGLE_THREAD_CARD_EXPAND_THUMBNAIL, {
+    comment: 'hello',
+    thread: 'foo',
+    subreddit: 'bar',
+  });
+  t.true(
+    t.context.state.getIn(['expandThumbnail', 'bar', 'foo']),
+    'should be toggled on since it was false by default'
+  );
+
+  t.context.state = fireAction(subreddits, t.context.state, SUBREDDITS.TOGGLE_THREAD_CARD_EXPAND_THUMBNAIL, {
```

Comment:
Is there a reason why the new state is being assigned to `t.context.state` rather than just a variable inside the test? Is it safe to do so?